### DISCUSSION TITLE
Error: data table is not defined

### DISCUSSION THREAD

**Discussion by ThorAlex87** [UtilitechAS/amsreader-firmware#410] [Q&A] [answered]
So I'm trying to set up my new pow-u+ at my new house, and I seem to not get data from the han port. "Data table is not defined" show up in a couple of the graphs. As far as I can tell I've set everything up correctly, and i've double checked the han port is open. Kaifa meter from tensio. Anyone know what could be the issue?
![Screenshot_20230126_182555_Firefox](https://user-images.githubusercontent.com/111507745/214907200-b8baf81e-314f-4fa9-87ba-415fe51d0e70.jpg)
![Screenshot_20230126_183208_Firefox](https://user-images.githubusercontent.com/111507745/214907342-5ddbbfec-eb49-4165-a58b-0336ae7d8b3b.jpg)

**Comment by ArnieO** [comment, marked as answer]
Parity: 8N1 on Kamstrup. 
(See the link "Known hardware configurations" 😉 )

**Reply by ThorAlex87** [reply]
But I have a Kaifa 😉

**Reply by ArnieO** [reply]
> But I have a Kaifa 😉

Ah... sorry, I read too quickly! Parity 8E1 is indeed correct on your meter. 

This is the issue:
We discovered yesterday afternoon that there is a snag when using the newest version Pow-U on Kaifa/Nuri meters. 

According to the Norwegian HAN-NVE standard, only pins 1 and 2 in the RJ45 socket are used. However, the unused lines are apparently terminated on the meter side in Kaifa/Nuri meters. This interferes with our duplication of HAN+ and HAN- signals on the Pow-U side RJ45 socket. This was implemented to facilitate connection to Mid-European meters that use an RJ12 socket with HAN signals on pins 3/4. 

I'm about to write a [blog post](https://amsleser.no/blog) explaining this better - but the quick fix is to carefully cut part of the cable lines of the RJ45 cable. There are 8 wires in the cable, you need to cut at least the 5 first of them (and no worries if you cut the 6th - but be careful not to cut one of the last two!)
![image](https://user-images.githubusercontent.com/10295178/215042670-bfe3c3ef-2ce4-4ed6-bfb8-a5aface48d57.png)

To be sure the cut lines do not come in contact later, you might want to do two cuts and remove a small piece of the 5 first lines.

**Reply by ThorAlex87** [reply]
Thanks! I have the rj45 tool so j guess I'll make a custom cable then, seems more elegant. 1 and 2 on the rj45 are the ones that need to be connected, correct?

**Reply by ArnieO** [reply]
> Thanks! I have the rj45 tool so j guess I'll make a custom cable then, seems more elegant. 1 and 2 on the rj45 are the ones that need to be connected, correct?

Exactly. 
And making a new cable is a good and more elegant option.

**Reply by ThorAlex87** [reply]
Thanks, new cable is made. I'll test it tomorrow when I'm up at the new place again.

**Reply by ArnieO** [reply]
Here is the full explanation and fix description:
https://amsleser.no/blog/post/27-i-have-a-kaifanuri-meter-and-my-new-pow-u-does-not-work

**Reply by ThorAlex87** [reply]
I hooked up the new cable today and everything worked perfectly!
